packages/financial-templates-lib/src/clients/OptimisticOracleClient.ts

```diff
@@ -154,32 +156,56 @@ export class OptimisticOracleClient {
 
   public async update(): Promise<void> {
     // Determine earliest block to query events for based on lookback window:
+    const netId = await this.web3.eth.getChainId();
```

Comment:
might it be better to send this in on construction?

Reply:
I don't love the idea of adding another constructor variable for something we'll only use once. I feel like its cleanest to infer `chainId` from the `web` object, even if that means an extra call. What we could do is store this `chainId` and only set it once

Reply:
Thoughts on this commit addressing your comment? https://github.com/UMAprotocol/protocol/pull/3742/commits/245bc900b8d7aeeb2c85c9e7013de602370b2f10

Reply:
your changes are better, thanks!